pkg/module/utils_test.go

```diff
@@ -57,3 +61,154 @@ func TestSubst(t *testing.T) {
+		var hostname, timestamp string
+		_, err = fmt.Sscanf(string(content), "%d %s %s", &pid, &hostname, &timestamp)
+		assert.NoError(t, err, "Should be able to parse lock file content")
+		assert.Equal(t, os.Getpid(), pid, "PID should match current process")
+
+		// Clean up
+		unlock()
+
+		// Verify lock file is removed
+		_, err = os.Stat(lockFile)
+		assert.Error(t, err, "Lock file should be removed")
+		assert.True(t, os.IsNotExist(err), "Lock file should not exist")
+	})
+
+	// Test 2: Cannot acquire lock when already locked
+	t.Run("Lock Contention", func(t *testing.T) {
+		unlock1, err := tryLockWithTimeout(tempDir, 1*time.Second)
+		assert.NoError(t, err, "Should successfully acquire first lock")
+		defer unlock1()
+
+		// Try to acquire second lock - should fail quickly
+		unlock2, err := tryLockWithTimeout(tempDir, 500*time.Millisecond) // Use shorter timeout
+		assert.Error(t, err, "Should fail to acquire second lock")
+		assert.Nil(t, unlock2, "Should not return unlock function")
+		assert.Contains(t, err.Error(), "timeout", "Error should mention timeout")
+	})
+
+	// Test 3: Can acquire lock after stale lock is removed
+	t.Run("Stale Lock", func(t *testing.T) {
+		lockFile := filepath.Join(tempDir, ".lilypad.lock")
+
+		// Create a stale lock file with non-existent PID
+		content := fmt.Sprintf("%d %s %s", 999999999, "test-host", time.Now().Format(time.RFC3339))
+		err := os.WriteFile(lockFile, []byte(content), 0600)
+		assert.NoError(t, err, "Should create stale lock file")
+
+		// Should be able to acquire lock despite stale file
+		unlock, err := tryLockWithTimeout(tempDir, 1*time.Second)
+		assert.NoError(t, err, "Should acquire lock after stale lock")
+		assert.NotNil(t, unlock, "Should return unlock function")
+
+		// Clean up
+		unlock()
+	})
+
+	// Test 4: Can acquire lock after old lock is removed (timeout based)
+	t.Run("Old Lock", func(t *testing.T) {
+		lockFile := filepath.Join(tempDir, ".lilypad.lock")
+
+		// Create an old lock file with valid PID but old timestamp
+		oldTime := time.Now().Add(-11 * time.Minute).Format(time.RFC3339)
+		content := fmt.Sprintf("%d %s %s", os.Getpid(), "test-host", oldTime)
+		err := os.WriteFile(lockFile, []byte(content), 0600)
+		assert.NoError(t, err, "Should create old lock file")
+
+		// Should be able to acquire lock despite old file
+		unlock, err := tryLockWithTimeout(tempDir, 1*time.Second)
+		assert.NoError(t, err, "Should acquire lock after old lock")
+		assert.NotNil(t, unlock, "Should return unlock function")
+
+		// Clean up
+		unlock()
+	})
+
+	// Test 5: Concurrent locking attempts should work correctly
+	t.Run("Concurrent Locking", func(t *testing.T) {
+		// Create multiple directories to avoid test interference
+		concurrentDir := filepath.Join(t.TempDir(), "concurrent")
+		err := os.MkdirAll(concurrentDir, 0755)
+		assert.NoError(t, err, "Should create concurrent test directory")
+
+		// This will track which goroutine got the lock
+		gotLock := make(chan int, 5)
```

Comment:
Might be good to put this `5` in a variable if we expect it to be the same used in the loop a couple lines below.